docs/reference/src/language/syntax/functions.md

```diff
@@ -14,27 +32,122 @@ Functions can be annotated with the `pure` keyword.
 This indicates that the function does not cause any side effects.
 More details can be found in the [Purity](../concepts/purity.md) chapter.
 
-By default, functions are private and cannot be accessed from external components.
+## Calling a function
+
+A function can be called without a target (like a function call in other languages) or with
+a target (like a method call in other languages):
+
+```slint,no-preview
+export component Example {
+    property <string> my-property: my-function(); // Called without a target
+    property <int> my-other-property: my_button.my-other-function(); // Called with a named target
+
+    pure function my-function() -> string {
+        return "result";
+    }
+
+    Text {
+        text: root.my-function(); // Called with a pre-defined name
+    }
+
+    my_button := Button {
+        pure function my-other-function() -> int {
+            return 42;
+        }
+    }
+}
+```
+
+The two types of calls function mostly the same. The only differences are:
+
+- Functions can be called without a target only from the element that defines them and its children.
+That means a function defined at the root of a component can be called from anywhere within that component
+while a function defined in an element can be called from that element and any child elements of it.
+- When a function is overridden, calling it with a target allows the caller to choose the exact version
+it wants to call (see Function Overriding below).
```

Comment:
In fact, this is the same scoping as properties and callback.

Reply:
Done: I mentioned this explicitly now.